canonical/ubuntu-pro-client
# Remove lock from the config module

## Why is this needed?
We are active working on making the config module more focused on only handling the actual Pro client configuration. Right now, it still handles some functionality that is not directly tied to the configuration.

This is an initial work that removed some unused data paths from the UAConfig class and also completely remove the lock handling from the config module 

## Test Steps
Guarantee that no test breaks after this change and that our lock mechanism is still working.
For example:

1. Launch a LXD container
2. Run `pro attach`
3. On another window, run `pro refresh`
4. Check that `pro refresh` will only start after `pro attach` has finished

## Checklist


## Does this PR require extra reviews?
